pkg/hintrunner/zero/zerohint_ec.go

```diff
@@ -175,3 +175,78 @@ func createNondetBigint3V1Hinter(resolver hintReferenceResolver) (hinter.Hinter,
 
 	return newNondetBigint3V1Hint(res), nil
 }
+
+func newFastEcAddAssignNewYHint() hinter.Hinter {
+	return &GenericZeroHinter{
+		Name: "FastEcAddAssignNewY",
+		Op: func(vm *VM.VirtualMachine, ctx *hinter.HintRunnerContext) error {
+			//> value = new_y = (slope * (x0 - new_x) - y0) % SECP_P
```

Comment:
Hmmm `SECP_P` should also be in the context right? Given its not imported. I've seen `PRIME` in the code too before but that doesn't need an import?

Reply:
I think this is the case, where you actually read SECP_P from the context

Reply:
I was going to say I didn't think that we needed to do that here, as you can see the previous hint block that was executed a few lines above, in the same function, and where SECP_P is imported right there. However, the only line between both hint blocks is calling a method that actually executes another hint, and looking at the code, SECP_P should remain the same, but I think the safest choice is to check the scope, specially for cases when is not that simple to determine if the value remains the same. wdyt?

Reply:
So I think the right answer is to always check the context. Everything except PRIME should be in the context unless the hint imports it itself. 

This also technically means that every variable that is created in a hint should be added to the scope. We just add the ones we think are needed down the line in the scope rn. I think thats fine. If we face errors because of some hint not adding a variable to scope we can fix it then.